source/retryable-reads/tests/unified/handshakeError.yml

```diff
@@ -0,0 +1,68 @@
+description: "retryable reads handshake failures"
+
+schemaVersion: "1.0"
+
+runOnRequirements:
+  - minServerVersion: "4.2"
+    topologies: [replicaset, sharded, load-balanced]
+    auth: true
+
+createEntities:
+  - client:
+      id: &client0 client0
+      useMultipleMongoses: false
+      observeEvents: [commandStartedEvent, connectionCheckOutStartedEvent]
+      uriOptions: { retryReads: true, retryWrites: true, maxPoolSize: 5 }
+  - database:
+      id: &database0 database0
+      client: *client0
+      databaseName: &databaseName retryable-handshake-tests
+  - collection:
+      id: &collection0 collection0
+      database: *database0
+      collectionName: &collectionName coll
+
+initialData:
+  - collectionName: *collectionName
+    databaseName: *databaseName
+    documents:
+      - { _id: 1, x: 11 }
+      - { _id: 2, x: 22 }
+      - { _id: 3, x: 33 }
+
+tests:
+  - description: "InsertOne succeeds after retryable handshake network error"
+    operations:
+      - name: failPoint # fail the next connection establishment
+        object: testRunner
+        arguments:
+          client: *client0
+          failPoint:
+            configureFailPoint: failCommand
+            mode: { times: 2 }
+            data:
+              failCommands: [saslContinue, dropCollection] # use saslContinue here to avoid SDAM errors
+              closeConnection: true
+      - name: dropCollection
```

Comment:
Why is `dropCollection` used here? This seems more like a test of a write operation than anything pertaining to `findOne`.

Also, since `findOne` technically isn't in the CRUD spec, would it make more sense to use `find` here?

Reply:
@ShaneHarvey  caught the issue with the failPoint creating a usable connection, if we also make dropCollection fail and hit that we can have the usable connection get discarded. Good point, I'll update to use find

https://github.com/mongodb/specifications/pull/1115#discussion_r782642103

Reply:
The reasoning for `dropCollection` sounds like it's worth a YAML comment for future reference.